static/app/components/replays/videoReplayer.tsx

```diff
@@ -148,6 +157,42 @@ export class VideoReplayer {
     return this._attachments[index];
   }
 
+  /**
+   * Fetches videos from attachments and adds them to the _videos dictionary.
+   * Loads the segment at the requested index.
+   */
+  protected fetchVideo(index: number | undefined): HTMLVideoElement | undefined {
+    if (typeof index === 'undefined') {
```

Comment:
Should we guard against index being fully out of bounds? ie: `index < 0 || index >= this._attachments.length`

Also, i think this would work if we compared to undefined directly:
```suggestion
    if (index === undefined) {
```

Reply:
yep we should! will add